src/fcast.c

```diff
@@ -15,24 +22,28 @@ SEXP fcast(SEXP lhs, SEXP val, SEXP nrowArg, SEXP ncolArg, SEXP idxArg, SEXP fil
     SET_VECTOR_ELT(ans, i, VECTOR_ELT(lhs, i));
   }
   // get val cols
+  bool some_fill = any_NA_int(nrows*ncols, idx);
```

Comment:
what are the performance implications of having to re-scan the full output for NA before applying `fill=`?

Reply:
My guess would be insignificant but I could run a performance test to verify.

Reply:
quick test:

```r
DT = CJ(a = letters, b = letters, c = letters, d = letters, e = LETTERS)
DT[, v := rnorm(.N)]

system.time(replicate(5, dcast(DT, a + b + c + d ~ e, value.var = 'v', fill = 0)))
## master
#    user  system elapsed 
#  14.877   1.437   8.186 
## branch fix5512 @ 3c7fb24483cbe05cf2db0638be8a71c8f5120299
#    user  system elapsed 
#  14.817   1.414   8.163 

## if there _are_ NA to fill
DT = DT[-sample(.N, 1e5)]

system.time(replicate(5, dcast(DT, a + b + c + d ~ e, value.var = 'v', fill = 0)))
## master
#    user  system elapsed 
#  14.093   1.558   7.674 
## branch fix5512 @ 3c7fb24483cbe05cf2db0638be8a71c8f5120299
#    user  system elapsed 
#  14.268   1.399   7.674 
```

So it looks like not a significant concern, though please to pipe in if you think that benchmark is not representative. Thanks!